### ISSUE TITLE
[Issue]: Unable to save library and scan fails upon deletion of said library

### ISSUE THREAD

**Issue by soultaco83** [jellyfin/jellyfin#11732]
### Please describe your bug

I am unable to save certain libraries in jellyfin. When I hit ok it spins without progressing. 
The libraries involved are one called Anime Subbed. This library is a TV show library

The other two libraries involved are set at movie libraries.

When I delete them and attempt a library scan I also get an error. 

### Reproduction Steps

Seems to be only 3 libraries at the moment. 
open library to edit
hit ok
No response on screen

### Jellyfin Version

10.9.0

### if other:

10.9.2

### Environment

```markdown
- OS: Unraid
- Linux Kernel: Linux 6.7.12-Unraid x86_64
- Virtualization: Docker
- Clients: Browser
- Browser: Chrome, Firefox, Edge
- FFmpeg Version: 6.0.1-6
- Playback Method: N/A
- Hardware Acceleration: QSV
- GPU Model: Arc A770
- Reverse Proxy: n/a
- Base URL: n/a
```

### Jellyfin logs

```shell
Jellyfin logs do not show event. for edit of library
 

jellyfin log when library is deleted and scan is attempted scan fails. Root folder is gone which makes sense because the library was delete so this entry should not exist in the database


[2024-05-19 03:23:42.399 -04:00] [ERR] [26] MediaBrowser.Controller.Entities.BaseItem: Error refreshing owned items for "/config/data/root/default/Anime Subbed"
System.IO.DirectoryNotFoundException: Could not find a part of the path '/config/data/root/default/Anime Subbed'.
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerableFactory.FileSystemInfos(String directory, String expression, EnumerationOptions options, Boolean isNormalized)
   at System.IO.DirectoryInfo.InternalEnumerateInfos(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at Emby.Server.Implementations.IO.ManagedFileSystem.GetFileSystemEntries(String path, Boolean recursive)
   at MediaBrowser.Controller.Providers.DirectoryService.<>c.<GetFileSystemEntries>b__5_0(String p, IFileSystem fileSystem)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd[TArg](TKey key, Func`3 valueFactory, TArg factoryArgument)
   at MediaBrowser.Controller.Providers.DirectoryService.GetFileSystemEntries(String path)
   at MediaBrowser.Controller.IO.FileData.GetFilteredFileSystemEntries(IDirectoryService directoryService, String path, IFileSystem fileSystem, IServerApplicationHost appHost, ILogger logger, ItemResolveArgs args, Int32 flattenFolderDepth, Boolean resolveShortcuts)
   at MediaBrowser.Controller.Entities.CollectionFolder.CreateResolveArgs(IDirectoryService directoryService, Boolean setPhysicalLocations)
   at MediaBrowser.Controller.Entities.CollectionFolder.GetFileSystemChildren(IDirectoryService directoryService)
   at MediaBrowser.Controller.Entities.BaseItem.RefreshMetadata(MetadataRefreshOptions options, CancellationToken cancellationToken)
[2024-05-19 03:23:42.404 -04:00] [ERR] [26] Emby.Server.Implementations.ScheduledTasks.TaskManager: Error executing Scheduled Task
System.IO.DirectoryNotFoundException: Could not find a part of the path '/config/data/root/default/Anime Subbed'.
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerableFactory.FileSystemInfos(String directory, String expression, EnumerationOptions options, Boolean isNormalized)
   at System.IO.DirectoryInfo.InternalEnumerateInfos(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at Emby.Server.Implementations.IO.ManagedFileSystem.GetFileSystemEntries(String path, Boolean recursive)
   at MediaBrowser.Controller.Providers.DirectoryService.<>c.<GetFileSystemEntries>b__5_0(String p, IFileSystem fileSystem)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd[TArg](TKey key, Func`3 valueFactory, TArg factoryArgument)
   at MediaBrowser.Controller.Providers.DirectoryService.GetFileSystemEntries(String path)
   at MediaBrowser.Controller.IO.FileData.GetFilteredFileSystemEntries(IDirectoryService directoryService, String path, IFileSystem fileSystem, IServerApplicationHost appHost, ILogger logger, ItemResolveArgs args, Int32 flattenFolderDepth, Boolean resolveShortcuts)
   at MediaBrowser.Controller.Entities.CollectionFolder.CreateResolveArgs(IDirectoryService directoryService, Boolean setPhysicalLocations)
   at MediaBrowser.Controller.Entities.CollectionFolder.RequiresRefresh()
   at MediaBrowser.Providers.Manager.MetadataService`2.RefreshMetadata(BaseItem item, MetadataRefreshOptions refreshOptions, CancellationToken cancellationToken)
   at MediaBrowser.Controller.Entities.BaseItem.RefreshMetadata(MetadataRefreshOptions options, CancellationToken cancellationToken)
   at Emby.Server.Implementations.Library.LibraryManager.ValidateTopLibraryFolders(CancellationToken cancellationToken, Boolean removeRoot)
   at Emby.Server.Implementations.Library.LibraryManager.PerformLibraryValidation(IProgress`1 progress, CancellationToken cancellationToken)
   at Emby.Server.Implementations.Library.LibraryManager.ValidateMediaLibraryInternal(IProgress`1 progress, CancellationToken cancellationToken)
   at Emby.Server.Implementations.ScheduledTasks.ScheduledTaskWorker.ExecuteInternal(TaskOptions options)
```

### FFmpeg logs

```shell
n/a
```

### Please attach any browser or client logs here

[10.10.10.20-1716103184321.log](https://github.com/jellyfin/jellyfin/files/15368515/10.10.10.20-1716103184321.log)
[log_20240519.log](https://github.com/jellyfin/jellyfin/files/15368524/log_20240519.log)

### Please attach any screenshots here
Browser console when attempting to edit library
![image](https://github.com/jellyfin/jellyfin/assets/76927195/4bb205e3-f85b-4392-8ec6-27b8c44903e2)

Root folder
![image](https://github.com/jellyfin/jellyfin/assets/76927195/6d848246-02f1-4d27-bc6b-a7921a3a9782)

the edit screen
![image](https://github.com/jellyfin/jellyfin/assets/76927195/6893643a-9703-45e3-b226-9992ac971967)

linux console to show location of folder after library delete
![image](https://github.com/jellyfin/jellyfin/assets/76927195/878f1349-9ed7-445b-84a3-06772937d307)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Simsala91** [comment]
Have the same issue, was able to fix library scans again by manually adding the folder /config/data/root/default/[Name of Deleted Library] with copies of the .mblink .xml and .collection files from a similar library (you might need to adjust some .xml data or names or get the files from a backup). That undeleted the library, but at least library scans worked again. 

Just some workaround if you need to get library scans working again.

**Comment by soultaco83** [comment]
Same I manually created the folder and set it to disabled in the options.xml

**Comment by soultaco83** [comment]
might be fixed with latest push in the 10.10 nightly? 
I am able to delete with no error. And the few that I could not edit not edit and save fine.
